src/axom/quest/detail/MarchingCubesImpl.hpp

```diff
@@ -0,0 +1,997 @@
+  {
+    // Time this due to potentially slow memory allocation
+    AXOM_PERF_MARK_FUNCTION("MarchingCubesImpl::initialize");
+    clearDomain();
+
+    SLIC_ASSERT(conduit::blueprint::mesh::topology::dims(dom.fetch_existing(
+                  axom::fmt::format("topologies/{}", topologyName))) == DIM);
+
+    m_mvu = axom::quest::MeshViewUtil<DIM, MemorySpace>(dom, topologyName);
+
+    m_bShape = m_mvu.getCellShape();
+    m_coordsViews = m_mvu.getConstCoordsViews(false);
+    if(!maskFieldName.empty())
+    {
+      m_maskView = m_mvu.template getConstFieldView<int>(maskFieldName, false);
+    }
+  }
+
+  AXOM_HOST void setDataParallelism(MarchingCubesDataParallelism dataPar) override
+  {
+    constexpr MarchingCubesDataParallelism autoPolicy =
+      std::is_same<ExecSpace, axom::SEQ_EXEC>::value
+      ? MarchingCubesDataParallelism::hybridParallel
+#if defined(AXOM_USE_OPENMP) && defined(AXOM_USE_RAJA)
+      : std::is_same<ExecSpace, axom::OMP_EXEC>::value
+        ? MarchingCubesDataParallelism::hybridParallel
+#endif
+        : MarchingCubesDataParallelism::fullParallel;
+
+    m_dataParallelism = dataPar;
+
+    if(m_dataParallelism == axom::quest::MarchingCubesDataParallelism::byPolicy)
+    {
+      m_dataParallelism = autoPolicy;
+    }
+  }
+
+  /*!
+    @brief Set the scale field name
+    @param fcnFieldName Name of nodal function is in dom
+  */
+  void setFunctionField(const std::string& fcnFieldName) override
+  {
+    m_fcnView = m_mvu.template getConstFieldView<double>(fcnFieldName, false);
+  }
+
+  void setContourValue(double contourVal) override
+  {
+    m_contourVal = contourVal;
+  }
+
+  /*!
+    @brief Implementation of virtual markCrossings.
+
+    Virtual methods cannot be templated, so this implementation
+    delegates to an implementation templated on DIM.
+  */
+  void markCrossings() override
+  {
+    AXOM_PERF_MARK_FUNCTION("MarchingCubesImpl::markCrossings");
+
+    m_caseIdsFlat.resize(m_mvu.getCellCount(), 0);
+    m_caseIdsFlat.fill(0);
+
+    // Choose caseIds stride order to match function stride order.
+    Indexer fcnIndexer(m_fcnView.strides());
+    m_caseIdsIndexer.initializeShape(m_bShape, fcnIndexer.slowestDirs());
+    m_caseIds = axom::ArrayView<std::uint16_t, DIM, MemorySpace>(
+      m_caseIdsFlat.data(),
+      m_bShape,
+      m_caseIdsIndexer.strides());
+    SLIC_ASSERT_MSG(Indexer(m_caseIds.strides()).getStrideOrder() ==
+                      fcnIndexer.getStrideOrder(),
+                    "Mismatched order is inefficient.");
+
+    markCrossings_dim();
+  }
+
+  //!@brief Populate m_caseIds with crossing indices.
+  template <int TDIM = DIM>
+  typename std::enable_if<TDIM == 2>::type markCrossings_dim()
+  {
+    MarkCrossings_Util mcu(m_caseIds, m_fcnView, m_maskView, m_contourVal);
+
+    auto order = m_caseIdsIndexer.getStrideOrder();
+    // order ^= axom::ArrayStrideOrder::BOTH; // Pick wrong ordering to test behavior.
```

Comment:
Do we still need this line, and (new) line 242?  Delete if not needed, please.

Reply:
Not needed.  Removing.  Thanks.